unison-cli/src/Unison/Codebase/Editor/HandleInput/DeleteBranch.hs

```diff
@@ -0,0 +1,67 @@
+-- | @delete.branch@ input handler
+module Unison.Codebase.Editor.HandleInput.DeleteBranch
+  ( handleDeleteBranch,
+  )
+where
+
+import Control.Lens (over, (^.))
+import qualified Data.Map.Strict as Map
+import Data.These (These)
+import qualified U.Codebase.Sqlite.Queries as Queries
+import Unison.Cli.Monad (Cli)
+import qualified Unison.Cli.Monad as Cli
+import qualified Unison.Cli.MonadUtils as Cli
+import qualified Unison.Cli.ProjectUtils as ProjectUtils
+import qualified Unison.Codebase.Branch as Branch
+import qualified Unison.Codebase.Editor.Output as Output
+import qualified Unison.Codebase.Path as Path
+import Unison.Prelude
+import Unison.Project (ProjectAndBranch (..), ProjectBranchName, ProjectName)
+import Witch (unsafeFrom)
+
+-- | Delete a project branch.
+--
+-- Currently, deleting a branch means deleting its `project_branch` row, then deleting its contents from the namespace.
+-- Its children branches, if any, are reparented to their grandparent, if any. You may delete the only branch in a
+-- project.
+handleDeleteBranch :: These ProjectName ProjectBranchName -> Cli ()
+handleDeleteBranch projectAndBranchTheseNames = do
+  projectAndBranchNames <- ProjectUtils.hydrateNames projectAndBranchTheseNames
+
+  maybeCurrentBranch <- ProjectUtils.getCurrentProjectBranch
+
+  deletedBranch <-
+    Cli.runEitherTransaction do
+      Queries.loadProjectBranchByNames (projectAndBranchNames ^. #project) (projectAndBranchNames ^. #branch) >>= \case
+        Nothing -> pure (Left (Output.LocalProjectBranchDoesntExist projectAndBranchNames))
+        Just branch -> do
+          Queries.deleteProjectBranch (branch ^. #projectId) (branch ^. #branchId)
+          pure (Right branch)
+
+  let projectId = deletedBranch ^. #projectId
+
+  Cli.stepAt
```

Comment:
The root branch update should probably happen in the same transaction as the project deletes yeah?

We may need to add a new combinator to actually allow us to do that, but I suspect we'll want that combinator in future project work anyways.

Reply:
Yeah, that's true. It should be fine to have the branch lingering in the namespace despite being deleted from sqlite, but one transaction would be even better.